mrmustard/lab/circuit_components.py

```diff
@@ -461,7 +461,8 @@ def quadrature(self, *quad: RealVector, phi: float = 0.0) -> ComplexTensor:
                 ret.shape[: -self.ansatz.batch_dims] if self.ansatz.batch_shape else ret.shape
             )
         )
-        return math.reshape(ret, (size,) + self.ansatz.batch_shape)
+        ret = math.reshape(ret, (size,) + self.ansatz.batch_shape)
+        return math.sum(ret, axis=-1) if self.ansatz._lin_sup else ret
```

Comment:
For my own understanding, why the need for a reshape and a sum here at the end? At first glance, one might think an expression like `self.to_quadrature(phi=phi).ansatz.eval(*quad, batch_string=batch_str)` would be sufficient.